Added support for ini file store

oldpatricka opened the pull request:
Hi there!

This PR adds support for ini files, in an effort to implement https://github.com/mozilla/sops/issues/70. It looks like https://github.com/mozilla/sops/pull/391 was an attempt to implement ini, but .env is distinct from .ini as far as I can see. For example, .env doesn't support sections.

I've attempted to encode the sops metadata in a way that will be readable in diffs, but this is a bit challenging for ini files since things like arrays aren't really supported.

I'm happy to take any suggestions if something looks off. :)

oldpatricka commented:
HI there, I've cleaned up the metadata encoding to match the dotenv implementation, so it should be ready to review. 👋

autrilla commented on stores/ini/store.go:
```diff
@@ -0,0 +1,321 @@
+)
+
+// Store handles storage of ini data.
+type Store struct {
```
Is this file `goimports`/`gofmt`ed? I feel like they'd move the braces to the same line.

oldpatricka replied:
Good, point. I was using an editor that doesn't automatically do this. I've run gofmt/goimports on the new files now.

oldpatricka commented:
Thanks for the review! I'll get to that this week!

oldpatricka commented:
@autrilla I've added an example file to the root of the directory, as well as a special tree for new ini files, since the existing example uses a tree structure with nested structures not supported by ini.

I've also added some clarification of filetypes supported to the README, but if you had something else in mind, I'd be happy to write it up.

autrilla commented:
Thanks a lot for taking the time to submit this patch!
